pydra/engine/core.py

```diff
@@ -882,6 +878,11 @@ def create_connections(self, task):
 
     async def _run(self, submitter=None, rerun=False, **kwargs):
         # self.inputs = dc.replace(self.inputs, **kwargs) don't need it?
+        # output_spec needs to be set using set_output or at workflow initialization
```

Comment:
i believe the only place to set an output_spec is via the init, so if any check should happen it should go to init, rather than in _run.

Reply:
output_spec is set at initialization, but is overwritten if you run `set_output` after adding all the tasks.  

```python
def set_output(self, connections):
...
self.output_spec = SpecInfo(name="Output", fields=fields, bases=(BaseSpec,))
logger.info("Added %s to %s", self.output_spec, self)
```

Reply:
Unless you passed in all the tasks to a workflow at workflow initialization already it doesn't actually make a lot of sense to specify output_spec...?   Let me know if I misunderstood anything